src/main/java/application/security/AuthenticationServiceImpl.java

```diff
@@ -16,9 +16,14 @@
     @Override
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> user = userService.findByLogin(login);
-        if (user.isEmpty() || !HashUtil.isValid(password, user.get())) {
+        if (user.isEmpty() || !isValid(password, user.get())) {
             throw new AuthenticationException("Incorrect username or password");
         }
         return user.get();
     }
+
+    private boolean isValid(String password, User user) {
+        byte[] salt = user.getSalt();
+        return HashUtil.hashPassword(password, salt).equals(HashUtil.hashPassword(password, salt));
```

Comment:
^^ it's a win-win situation) let's reconsider this line

Reply:
done...